src/lib/storage/segment_iterate.hpp

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include "pos_list.hpp"
+#include "resolve_type.hpp"
+#include "storage/base_segment.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/segment_iterables.hpp"
+#include "storage/segment_iterables/any_segment_iterable.hpp"
+
+/**
+ * This file provides the main entry points to read Segment data, irrespective of the underlying encoding.
+ *
+ * Two main signatures are provided:
+ *      segment_with_iterators[_filtered]()    Calls the functor with a begin and end iterator
+ *      segment_iterate[_filtered]()           Calls the functor with each value in the segment
+ *
+ * Both functions optionally take a PosList which allows for selective access to the values in a segment.
+ *
+ * The template parameter T is either (if known to the caller) the DataType of the values contained in the segment, or
```

Comment:
Why would the caller pass T?

Reply:
Because he calls the function from code that is already instantiated per DataType

Reply:
See, e.g., _materialize_sort_column

Reply:
My fault, I was not clear enough. I would add a comment saying why it would be good for the caller to do this.